lib/spack/spack/schema/cdash.py

```diff
@@ -14,14 +14,28 @@
     "cdash": {
         "type": "object",
         "additionalProperties": False,
-        # "required": ["build-group", "url", "project", "site"],
-        "required": ["build-group"],
+        "required": ["upload-url"],
         "patternProperties": {
-            r"build-group": {"type": "string"},
-            r"url": {"type": "string"},
-            r"project": {"type": "string"},
+            r"buildstamp": {"type": "string"},
```

Comment:
This is a breaking configuration change, the field needs to go through a deprecation before it can just be dropped/renamed.

Also anything we change here needs to be reflected in `lib/spack/spack/reporters/cdash.py`, `lib/spack/spack/ci.py::CDashHandler` and in all of the CDash configurations used under `share/spack/gitlab/cloud_pipelines/`

---

From what I can tell, `build-group` and `buildstamp` are potentially related, but not required. `buildstamp` is a dynamic field, the default is to compose it from the `build-group` and the current timestamp. I don't think it makes sense in the config here.

`build-name` is another dynamic field that is usually computed by CI, not statically configured

`project` is a thing in CDash. I think that should be kept here.

`upload-url` is not a 1-1 for `url`, I also don't think it is convenient or readable in the config. Note, `upload-url` is something like `my.cdash.org/submit.php?project=MyProject` where as `url` is just `my.cdash.org`. Having to compose the API request in the config does not seem like an improvement.

`track` makes sense to me in the config.

@zackgalbreath is the cdash expert, so maybe he has some other ideas about naming here as well.

Reply:
Thanks for pointing out where this is used, @kwryankrattiger.  I missed that in grepping.  When I get back to this I will look into implementing in a way that maintains the current API.